store/storetest/channel_store.go

```diff
@@ -563,36 +563,25 @@ func testChannelStoreGetByName(t *testing.T, ss store.Store) {
 	o1.Type = model.CHANNEL_OPEN
 	store.Must(ss.Channel().Save(&o1, -1))
 
-	r1 := <-ss.Channel().GetByName(o1.TeamId, o1.Name, true)
-	if r1.Err != nil {
-		t.Fatal(r1.Err)
-	} else {
-		if r1.Data.(*model.Channel).ToJson() != o1.ToJson() {
-			t.Fatal("invalid returned channel")
-		}
-	}
+	result := <-ss.Channel().GetByName(o1.TeamId, o1.Name, true)
+	require.Nil(t, result.Err)
+	require.Equal(t, o1.ToJson(), result.Data.(*model.Channel).ToJson(), "invalid returned channel")
 
-	if err := (<-ss.Channel().GetByName(o1.TeamId, "", true)).Err; err == nil {
-		t.Fatal("Missing id should have failed")
-	}
+	channelID := result.Data.(*model.Channel).Id
 
-	if r1 := <-ss.Channel().GetByName(o1.TeamId, o1.Name, false); r1.Err != nil {
-		t.Fatal(r1.Err)
-	} else {
-		if r1.Data.(*model.Channel).ToJson() != o1.ToJson() {
-			t.Fatal("invalid returned channel")
-		}
-	}
+	result = <-ss.Channel().GetByName(o1.TeamId, "", true)
+	require.NotNil(t, result.Err, "Missing id should have failed")
 
-	if err := (<-ss.Channel().GetByName(o1.TeamId, "", false)).Err; err == nil {
-		t.Fatal("Missing id should have failed")
-	}
+	result = <-ss.Channel().GetByName(o1.TeamId, o1.Name, false)
+	require.Nil(t, result.Err)
+	require.Equal(t, o1.ToJson(), result.Data.(*model.Channel).ToJson(), "invalid returned channel")
 
-	store.Must(ss.Channel().Delete(r1.Data.(*model.Channel).Id, model.GetMillis()))
+	result = <-ss.Channel().GetByName(o1.TeamId, "", false)
+	require.NotNil(t, result.Err, "Missing id should have failed")
 
-	if err := (<-ss.Channel().GetByName(o1.TeamId, r1.Data.(*model.Channel).Name, false)).Err; err == nil {
-		t.Fatal("Deleted channel should not be returned by GetByName()")
-	}
+	store.Must(ss.Channel().Delete(channelID, model.GetMillis()))
+	result = <-ss.Channel().GetByName(o1.TeamId, channelID, false)
```

Comment:
This isn't the same assertion as before: the test used to verify that fetching a deleted channel by name didn't return anything, but we're not passing in a `channelID` against which to compare which should, of course, always yield nothing.

Can we just use `o1.Name` directly here?

Reply:
Good catch!